### PR TITLE
Chi 2023 warm transfer reservation timeout stuck state

### PR DESCRIPTION
**Pull request by acedeywin**
## Description

- Bug fix: Warm transfer reservation timeout causes a stuck state

**Problem**: During a warm transfer, if the receiving counsellor doesn’t action the transfer reservation, the reservation will time out, but the form and transfer state will remain locked for the original counsellor.

**Solution**: Added a check to ascertain whether warm voice transfer timed out and updated the task attribute.

### Checklist
- [x] Corresponding issue has been opened
- [N/A] New tests added

### Related Issues
Fixes CHI-2023

### Verification steps

- Login to Aselo as two counsellors. I did this using my Okta login (A) and Twilio login (B).
- Call the Aselo Dev number
- A picks up the call, and WARM transfers (consult) to B.
- B ignore the call so that it will time out. The transfer should time out after 2 minutes.
- Verify that the original counsellor gets the task back in the original state (form unlocked, ability to transfer and conference).

### CODE REVIEW

**Comment by stephenhand** [comment]
This branch is based off master, it needs to be off `v2.8-rc` if we want to go out in a v2.8.x patch

**Comment by acedeywin** [comment]
> This branch is based off master, it needs to be off `v2.8-rc` if we want to go out in a v2.8.x patch

@stephenhand  - I can't find any v2.8-rc branch.

**Comment by stephenhand** [comment]
Sorry `v1.11-rc`

**Review by stephenhand — APPROVED** [review]
Some minor points around logging but looks good otherwise and the fix works in my tests

File: functions/conference/getParticipant.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
```
[the remark this answers is not in the document]
Comment (acedeywin) [reply functions/conference/getParticipant.ts:1]: @stephenhand  - I pull from the master before committing—that's where this is from.

File: functions/taskrouterListeners/transfersListener.private.ts [hunk 2]
```diff
@@ -132,6 +179,8 @@ export const handleEvent = async (context: Context<EnvVars>, event: EventFields)
 
     const taskAttributes = JSON.parse(taskAttributesString);
 
+    console.log(isWarmVoiceTransferTimedOut(eventType, taskChannelUniqueName, taskAttributes));
```
[the remark this answers is not in the document]
Comment (acedeywin) [reply functions/taskrouterListeners/transfersListener.private.ts:182]: Done.

File: functions/taskrouterListeners/transfersListener.private.ts [hunk 3]
```diff
@@ -113,6 +113,53 @@ const isVoiceTransferOriginalInWrapup = (
...
+ *    - transferMeta.sidWithTaskControl: to original reservation
+ * Same applies to when transfer timeout
+ */
+const updateWarmVoiceTransferAttributes = async (
+  transferStatus: string,
+  context: Context<EnvVars>,
+  taskAttributes: { transferMeta: { originalReservation: string } },
+  taskSid: string,
+) => {
+  console.log(`Handling warm voice transfer ${transferStatus}...`);
```
[the remark this answers is not in the document]
Comment (acedeywin) [reply functions/taskrouterListeners/transfersListener.private.ts]: @stephenhand  - Noted. I have updated the console logs to console info with the taskSid inclusive.

**Pull request merged**